best_practices/api_call_sequencing.md

```diff
@@ -0,0 +1,51 @@
+---
+permalink: /api-call-sequencing
+title: App Installation Design and API Call Sequencing
+layout: default
+section_title: Best Practices
+
+---
+
+## Background ##
+
+As you develop your project, it is important to understand that your design choices and the order in which API calls are made can have a significant impact on the behavior and performance of your application.
+Proper installation design and API call sequencing can help prevent unintended effects when a user with access to more than one company authorizes your application through OAuth 2.0.
+
+## Installation with OAuth 2.0 Authorization
+
+An application not installed from the App Marketplace by a Procore Admin may still be accessed and used by a Procore user if they authorize the application to access their Procore data through the OAuth 2.0 flow.
+The ability to authorize and use an application in a Procore account is controlled by the [Allow User Installs](https://support.procore.com/products/online/user-guide/company-level/admin/tutorials/allow-user-installs) feature in the Procore Web application.
+When a user authorizes the application and subsequently accesses Procore API resources in this manner, it is connected to and visible in the company level App Management tool in Procore.
+If Procore users are able to authorize your application through OAuth 2.0 outside the context of a Marketplace installation, it is important to account for this scenario in your design.
```

Comment:
I didn't understand this paragraph. 

What is the Best Practice that a developer is meant to follow?

Reply:
The last sentence.

Reply:
I still don't understand what a developer should change when creating an app to "account for this scenario".

As far as I know, there's only one way to create an app, and that's through this editor:

<img width="647" alt="Snap 2021-06-16 at 2 33 03 PM" src="https://user-images.githubusercontent.com/74197530/122447881-0ce05700-cf59-11eb-8aea-54c84b45a44c.png">

Reply:
It is meant to be a general, conceptual statement informing the developer that they need to be aware of the `Allow User Installs' option and how it affects installations, so that they can plan appropriately during their design phase.